backend/multimodalexplorer/functions/search_faiss_index.py

```diff
@@ -87,7 +88,31 @@ def _process_search_query(self, search_query: dict) -> np.ndarray:
 
         return query_embedding.numpy().astype(np.float32)
 
-    def search_index(self, search_query: dict) -> list:
+    def _query_result(self, indices) -> list:
```

Comment:
in line 79 above. 
```
data2vec_model = load_model(search_type)
```
I think you want to load the model once, not for each request.